src/lib/tls/credentials_manager.cpp

```diff
@@ -21,10 +22,60 @@ std::string Credentials_Manager::psk_identity(const std::string& /*unused*/,
    return "";
 }
 
-SymmetricKey Credentials_Manager::psk(const std::string& /*unused*/,
-                                      const std::string& /*unused*/,
+SymmetricKey Credentials_Manager::psk(const std::string& type,
+                                      const std::string& context,
                                       const std::string& identity) {
-   throw Internal_Error("No PSK set for identity " + identity);
+   auto side = [&] {
+      if(type == "tls-client") {
+         return TLS::Connection_Side::Client;
+      } else if(type == "tls-server") {
+         return TLS::Connection_Side::Server;
+      } else {
+         throw Internal_Error(fmt("No PSK set for type {}", type));
+      }
+   }();
+
+   // New applications should use the appropriate credentials methods. This is a
+   // retrofit of the behaviour before Botan 3.2.0 and will be removed in a
+   // future major release.
+   //
+   // TODO: deprecate `psk("...", "session-ticket" | "dtls-cookie-secret")`
+   if(side == TLS::Connection_Side::Server && context == "session-ticket") {
+      if(auto key = session_ticket_key(); !key.empty()) {
+         return SymmetricKey(std::move(key));
+      }
+   } else if(side == TLS::Connection_Side::Server && context == "dtls-cookie-secret") {
+      if(auto key = dtls_cookie_secret(); !key.empty()) {
+         return SymmetricKey(std::move(key));
+      }
+   } else /* context is a host name */ {
+      // Assuming that find_preshared_keys returns _exactly_ one or no keys when
+      // searching for a single specific identity.
+      if(auto psks = find_preshared_keys(context, side, {std::string(identity)}); psks.size() == 1) {
```

Comment:
```suggestion
      if(auto psks = find_preshared_keys(context, side, {identity}); psks.size() == 1) {
```

... artefact from trying to move to `std::string_view` method parameter types.